dir.go

```diff
@@ -117,13 +114,13 @@ func (d *Directory) localUpdate(c child) (*dag.ProtoNode, error) {
 		return nil, dag.ErrNotProtobuf
 	}
 
-	err = d.dagService.Add(d.ctx, nd)
```

Comment:
is it worth having context still an argument to constructor and threaded through to things like this, but not saved in the struct?

Reply:
I have been considering this, and where the use of context was removed and replaced with `context.Background()` are the places that I thought it was better to not support cancellation at all.

Suppose there is a single request that creates a new directory, puts some data into that directory, deletes the data from some other place.   Do we really want cancellation in the middle of this?  If a request is cancelled or timed out due to a stall in a lower level (e.g. datastore), it may better to let the request continue and either eventually succeed or error out.  Contexts spread like viruses, so I was hoping to limit the spread to only what it really necessary. 

@Stebalien Do you see there is a need to thread contexts through the APIs?  Is so I will make that change.

Reply:
This should take a context (from Flush). Flush _should_ support cancellation as it might need to write a lot of data (and that update may need to be canceled).